### PR TITLE
Formatting, pvc gemm args validation, onemkl cmake

### PR DESCRIPTION
**Pull request by FMarno**
A couple changes I noticed while looking into something.

### CODE REVIEW

File: cmake/onemkl.cmake [hunk 1]
```diff
@@ -30,6 +30,8 @@ include_guard()
 
 include(ExternalProject)
 
+option(CUTLASS_SYCL_DISCONNECT_ONEMKL_UPDATE "Option is passed to the UPDATE_DISCONNECTED parameter of ExternalProject_Add for onemkl" YES)
```
[the remark this answers is not in the document]
Comment (FMarno) [reply cmake/onemkl.cmake]: done, watcha think?

File: cmake/onemkl.cmake [hunk 2]
```diff
@@ -57,6 +59,7 @@ ExternalProject_Add(
     -DTARGET_DOMAINS=rng
     INSTALL_DIR ${ONEMKL_INSTALL_DIR}
     BUILD_BYPRODUCTS ${ONEMKL_LIB}
+    UPDATE_DISCONNECTED ${CUTLASS_SYCL_DISCONNECT_ONEMKL_UPDATE}
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply cmake/onemkl.cmake:62]: Assuming this doesn't break a fresh build (CMake should be clever enough to fetch the first time, right?) then it seems fine :+1:

File: include/cutlass/gemm/kernel/xe_gemm.hpp [hunk 3]
```diff
@@ -153,6 +153,14 @@ class GemmUniversal<
 
   static bool
   can_implement(Arguments const& args) {
+    auto m = get<0>(args.problem_shape);
+    auto n = get<1>(args.problem_shape);
+    auto k = get<2>(args.problem_shape);
+    bool m_valid = m > 0;
+    bool n_valid = n > 0 && n % 4 == 0;
+    bool k_valid = k > 0 && k % get<2>(TileShape{}) == 0;
+    if (!(m_valid and n_valid and k_valid)) return false;
```
Comment (joeatodd) [inline include/cutlass/gemm/kernel/xe_gemm.hpp:162]: Total nit so please feel free to ignore:
```suggestion
    shape_implementable =  (m_valid and n_valid and k_valid);
```
& then add it to the return statement. Slightly neater (?) but with potentially redundant checks which your solution short-circuits.

**Review by joeatodd — APPROVED** [review]
LGTM - might be worth confirming the `UPDATE_DISCONNECTED` won't break a fresh build.

**Review by t4c1 — APPROVED** [review]

**Review by aacostadiaz — APPROVED** [review]
Looks good!

**Pull request merged**